### PR TITLE
Migrate `invite_new` protocol tests

### PR DESCRIPTION
**Pull request by FirelightFlagboy**

### CODE REVIEW

File: libparsec/crates/protocol/tests/authenticated_cmds/v3/invite_new.rs [hunk 1]
```diff
@@ -1,27 +1,147 @@
...
+fn rep_helper(raw: &[u8], expected: Result<authenticated_cmds::invite_new::Rep, String>) {
+    let res = authenticated_cmds::invite_new::Rep::load(raw).map_err(|e| e.to_string());
+
+    p_assert_eq!(res, expected);
+
+    if let Ok(data) = (res) {
+        // Also test serialization round trip
+        let raw2 = data.dump().unwrap();
+
+        let data2 = authenticated_cmds::invite_new::Rep::load(&raw2).map_err(|e| e.to_string());
```
Comment (TimeEngineer) [inline libparsec/crates/protocol/tests/authenticated_cmds/v3/invite_new.rs:143]: don't map to string

Comment (FirelightFlagboy) [reply libparsec/crates/protocol/tests/authenticated_cmds/v3/invite_new.rs:143]: `::rpm_serde::decode::Error` don't implement `PartialEq` & `Eq` do you have a suggestion to circumvent that ?

Comment (TimeEngineer) [reply libparsec/crates/protocol/tests/authenticated_cmds/v3/invite_new.rs:143]: Don't group error case with good case

Comment (FirelightFlagboy) [reply libparsec/crates/protocol/tests/authenticated_cmds/v3/invite_new.rs]: done

**Review by TimeEngineer — APPROVED** [review]

**Pull request merged**